charts/k8s-job/templates/_job_spec.tpl

```diff
@@ -0,0 +1,200 @@
+{{- end -}}
+{{- $allContainerPorts := values .Values.containerPorts -}}
+{{- range $allContainerPorts -}}
+  {{/* We are exposing ports if there is at least one key in containerPorts that is not disabled (disabled = false or
+       omitted)
+  */}}
+  {{- if or (not (hasKey . "disabled")) (not .disabled) -}}
+    {{- $_ := set $hasInjectionTypes "exposePorts" true -}}
+  {{- end -}}
+{{- end -}}
+{{- $allSecrets := values .Values.secrets -}}
+{{- range $allSecrets -}}
+  {{- if eq (index . "as") "volume" -}}
+    {{- $_ := set $hasInjectionTypes "hasVolume" true -}}
+  {{- else if eq (index . "as") "environment" -}}
+    {{- $_ := set $hasInjectionTypes "hasEnvVars" true -}}
+  {{- else if eq (index . "as") "envFrom" }}
+    {{- $_ := set $hasInjectionTypes "hasEnvFrom" true -}}
+  {{- else if eq (index . "as") "none" -}}
+    {{- /* noop */ -}}
+  {{- else -}}
+    {{- fail printf "secrets config has unknown type: %s" (index . "as") -}}
+  {{- end -}}
+{{- end -}}
+{{- $allConfigMaps := values .Values.configMaps -}}
+{{- range $allConfigMaps -}}
+  {{- if eq (index . "as") "volume" -}}
+    {{- $_ := set $hasInjectionTypes "hasVolume" true -}}
+  {{- else if eq (index . "as") "environment" -}}
+    {{- $_ := set $hasInjectionTypes "hasEnvVars" true -}}
+  {{- else if eq (index . "as") "envFrom" }}
+    {{- $_ := set $hasInjectionTypes "hasEnvFrom" true -}}
+  {{- else if eq (index . "as") "none" -}}
+    {{- /* noop */ -}}
+  {{- else -}}
+    {{- fail printf "configMaps config has unknown type: %s" (index . "as") -}}
+  {{- end -}}
+{{- end -}}
+{{- if gt (len .Values.persistentVolumes) 0 -}}
+  {{- $_ := set $hasInjectionTypes "hasVolume" true -}}
+{{- end -}}
+{{- if gt (len .Values.scratchPaths) 0 -}}
+  {{- $_ := set $hasInjectionTypes "hasVolume" true -}}
+{{- end -}}
+{{- if gt (len .Values.emptyDirs) 0 -}}
+  {{- $_ := set $hasInjectionTypes "hasVolume" true -}}
+{{- end -}}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "k8s-job.fullname" . }}
+  labels:
+    # These labels are required by helm. You can read more about required labels in the chart best practices guide:
+    # https://docs.helm.sh/chart_best_practices/#standard-labels
+    helm.sh/chart: {{ include "k8s-job.chart" . }}
+    app.kubernetes.io/name: {{ include "k8s-job.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    {{- range $key, $value := .Values.additionalJobLabels }}
+    {{ $key }}: {{ $value }}
+    {{- end}}
+{{- with .Values.jobAnnotations }}
+  annotations:
+{{ toYaml . | indent 4 }}
+{{- end }}
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "k8s-job.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "k8s-job.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        gruntwork.io/deployment-type: main
```

Comment:
This label is not necessary in `k8s-job`, as it is primarily used to differentiate with canaries, which doesn't apply for the job use case.